### DISCUSSION TITLE
how to setup keybinds to not introduce delay in vim normal mode

### DISCUSSION THREAD

**Discussion by schpet** [zed-industries/zed#19610] [Feature Requests]
given i'm interested in having `<leader>p` format my code, and `<leader>` to be <kbd>space</kbd> i've setup my keymap with this:

```jsonc
[
  {
    "context": "EmptyPane || SharedScreen || vim_operator == none && !VimWaiting && vim_mode != insert",
    "bindings": {
      "space p": "editor::Format",
      // and a bunch of other things mapped to space... 
    }
  }
]
```

but now any motion around `space` introduces delay. e.g. `R<space>` takes >500ms to replace with a space. `dt<space>` takes a similar delay to delete until a space.

note that the context is copied from https://zed.dev/blog/zed-decoded-vim – can that be tweaked to fix this? or is this a limitation that needs to be worked around or fixed in a different way?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
